src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs

```diff
@@ -163,13 +164,22 @@ public static IServiceCollection AddMvc([NotNull] this IServiceCollection servic
         internal static void AddMvcServices(IServiceCollection services)
         {
             // Options - all of these are multi-registration
-            services.AddTransient<IConfigureOptions<MvcOptions>, MvcOptionsSetup>();
-            services.AddTransient<IConfigureOptions<MvcOptions>, JsonMvcOptionsSetup>();
-            services.AddTransient<IConfigureOptions<MvcFormatterMappingOptions>, JsonMvcFormatterMappingOptionsSetup>();
-            services.AddTransient<IConfigureOptions<MvcViewOptions>, MvcViewOptionsSetup>();
-            services.AddTransient<IConfigureOptions<RazorViewEngineOptions>, RazorViewEngineOptionsSetup>();
-
-            services.TryAdd(ServiceDescriptor.Transient<IAssemblyProvider, DefaultAssemblyProvider>());
+            TryAddMultiRegistrationService(
+                services,
+                ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, MvcOptionsSetup>());
+            TryAddMultiRegistrationService(
+                services,
+                ServiceDescriptor.Transient<IConfigureOptions<MvcOptions>, JsonMvcOptionsSetup>());
+            TryAddMultiRegistrationService(
+                services,
+                ServiceDescriptor.Transient<IConfigureOptions<MvcFormatterMappingOptions>, JsonMvcFormatterMappingOptionsSetup>());
+            TryAddMultiRegistrationService(
+                services,
+                ServiceDescriptor.Transient<IConfigureOptions<MvcViewOptions>, MvcViewOptionsSetup>());
+            TryAddMultiRegistrationService(
+                services,
+                ServiceDescriptor
+                .Transient<IConfigureOptions<RazorViewEngineOptions>, RazorViewEngineOptionsSetup>());
```

Comment:
Odd to wrap this line and not the one for `JsonMvcFormatterMappingOptionsSetup`.